azerothcore/azerothcore-wotlk
# fix (core/logging): log money

## Changes Proposed:
-  Correction of texts in the topic and the type of logged action. Getting rid of `<>` tags, which can cause certain difficulties when processing logs by the administrator. 
-  Now we can sort entries by the type of logged actions by `type` column if need

## Tests Performed:

- Tested in game by sending gold, trading between 2 characters, deposit and wifdraw gold from/to Guildbank and sending by mail.



## How to Test the Changes:


1. make transactions with money with amount about 100 gold betwen characters by trade or mail, deposit to and withdraw from  GuildBank or receive profit from AuctionHouse. Type of action must be placed in `type` field of `log_money` table. 
2. If you deposit money to GB - check that field `receiver_acc` get the Guild ID and `receiver_name` get Guild Name instead 0 as it was before.
3. If you wifdraw money from GB - check that sender is Guild and receiver is character instead 0 as it was before.
4. Now you can process logged actions with filtering by action type. If in PHP - you don't need to parce `topic` field and make `htmlspecialchars($topic)`-clearing to obtain recorded action type. Just filter or search by `topic` column and you'll find what you want.


For updating `log_money` table on live server you can use this function via php:

[fix_live_table_php.txt](https://github.com/azerothcore/azerothcore-wotlk/files/7918282/fix_live_table_php.txt)





## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
